engine/docker/docker.go

```diff
@@ -223,6 +225,13 @@ func (e *Docker) Destroy(ctx context.Context, pipelineConfig *spec.PipelineConfi
 	return nil
 }
 
+func (e *Docker) IsDockerEnabled(pipelineConfig *spec.PipelineConfig) bool {
```

Comment:
This doesn't belong here. This layer just does docker operations - it doesn't check whether docker is installed or not. That should be whoever invokes this code.